### PR TITLE
Clean up/refactor GraphEdit

### PR DESCRIPTION
**Pull request by Geometror**
Split from #67152. Depends on #79307.
This PR contains the refactoring and cleanup of GraphEdit.

Detailed changes:
- Rename lots of variables to be more descriptive (e.g. icofs -> icon_offset, c -> child)
- Restructure the header and source files to follow a consistent scheme (i. a. order of member variable and method declarations, which were sometimes mixed)
- Fix comment style in various places
- Rename `get_zoom_hbox` to `get_menu_hbox`
- Rename `set/get_scroll_ofs` to `set/get_scroll_offset`
- Separate snapping and grid (https://github.com/godotengine/godot-proposals/discussions/5250)
	- separate properties and toggle buttons (use_snap and show_grid):
Pasted image 20221028165043
- `GraphEdit`'s node arrangement logic is now extracted into its own class `GraphEditArranger`; its coupling was loose enough so that only trivial adjustments were needed for it to use `GraphEdit`'s API

### CODE REVIEW

**Comment by YuriSizov** [comment]
I've rebased after https://github.com/godotengine/godot/pull/79307 and updated some names which were missed (or ignored?) from the previous review. I've also added the missing icons for the new toggle (took them from https://github.com/godotengine/godot/pull/67152), and renamed theme items to follow the general naming scheme (something I avoided during https://github.com/godotengine/godot/pull/65437, but now we can do it).

File: doc/classes/GraphEdit.xml [hunk 1]
```diff
@@ -30,8 +30,8 @@
 				Below is a sample code to help get started:
 				[codeblock]
 				func _is_in_input_hotzone(in_node, in_port, mouse_position):
-				    var port_size: Vector2 = Vector2(get_theme_constant("port_grab_distance_horizontal"), get_theme_constant("port_grab_distance_vertical"))
-				    var port_pos: Vector2 = in_node.get_position() + in_node.get_connection_input_position(in_port) - port_size / 2
+				    var port_size : Vector2 = Vector2(get_theme_constant("port_grab_distance_horizontal"), get_theme_constant("port_grab_distance_vertical"))
```
Comment (AThousandShips) [inline doc/classes/GraphEdit.xml:33]: This violates the [style-guide](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#declared-types)

**Review by YuriSizov — APPROVED** [review]
LGTM!

**Comment by YuriSizov** [comment]
Thanks!

**Comment by clayjohn** [comment]
This change is triggering a bunch of errors in the CI for breaking API compatibility. We need to add compatibility code or revert the API-breaking changes

**Comment by YuriSizov** [comment]
@clayjohn See https://github.com/godotengine/godot/pull/79307.

**Pull request merged**